test/geometry_tests.cpp

```diff
@@ -61,105 +94,176 @@ bool checkFirstOrLastPoint(int firstOrLastPoint, int x_shouldbe, int y_shouldbe,
 
 	// Check if the last point is (5,0)
-	ASSERT_EQ(checkFirstOrLastPoint(1, 5, 0, v), true);
+	// EXPECT_THAT(checkFirstOrLastPoint(1, 5, 0, v), testing::IsTrue());
+	EXPECT_THAT(v.back(), testing::ElementsAre(5,0));
 }
 
 TEST(Point2DPlanning_Griz_Test, bresenham_conversion_horz_line_test) {
 	// Create a horizontal line between points (0,0) and (5,0)
-	auto v = bresenham_conversion(0, 0, 5, 0);
+	const auto v1 = bresenham_conversion(0, 0, 5, 0);
 		
 	// Check there are 6 points
-	ASSERT_EQ(v.size(), 6);
+	EXPECT_THAT(v1.size(), testing::Eq(6));
 
 	// Check that all the y coordinates are the same (indicates horz line)
-	ASSERT_EQ(allXOrYCoorSame(1,v), true);	
+	EXPECT_THAT(allXOrYCoorSame(1,v1), testing::IsTrue());	
 
 	// Create a horizontal line from (4,5) to (-1,5)
-	v = bresenham_conversion(4, 5, -1, 5);
+	const auto v2 = bresenham_conversion(4, 5, -1, 5);
 	
 	// Check there are 6 points
-	ASSERT_EQ(v.size(), 6);	
+	EXPECT_THAT(v2.size(), testing::Eq(6));	
 
 	// Check that all the y coordinates are the same (indicates horz line)
-	ASSERT_EQ(allXOrYCoorSame(1,v), true);	
+	EXPECT_THAT(allXOrYCoorSame(1,v2), testing::IsTrue());	
 }
 
 TEST(Point2DPlanning_Griz_Test, bresenham_conversion_vert_line_test) {
 	// Create a vertical line from (-2,-3) to (-2, 2)
-	auto v = bresenham_conversion(-2,-3,-2,2);	
+	const auto v1 = bresenham_conversion(-2,-3,-2,2);	
 
 	// Check there are 6 points
-	ASSERT_EQ(v.size(), 6);
+	EXPECT_THAT(v1.size(), testing::Eq(6));
 
 	// Check that all of the x coordinates are the same (indicates vertical line)
-	ASSERT_EQ(allXOrYCoorSame(0,v), true);	
+	EXPECT_THAT(allXOrYCoorSame(0,v1), testing::IsTrue());	
 
-	v = bresenham_conversion(1, 2, 1, -3);
+	const auto v2 = bresenham_conversion(1, 2, 1, -3);
 	
 	// Check that there are 6 points
-	ASSERT_EQ(v.size(), 6);	
+	EXPECT_THAT(v2.size(), testing::Eq(6));	
 
 	// Check that all of the x coordinates are the same (indicates vertical line)
-	ASSERT_EQ(allXOrYCoorSame(0,v), true);	
+	EXPECT_THAT(allXOrYCoorSame(0,v2), testing::IsTrue());	
 }
 
 TEST(Point2DPlanning_Griz_Test, bresenham_conversion_point_test) {
 	// Get the points for the line connecting (0,0) and (0,0)
-	auto v = bresenham_conversion(0, 0, 0, 0);
+	const auto v1 = bresenham_conversion(0, 0, 0, 0);
 
 	// Check only one point was produced
-	ASSERT_EQ(v.size(), 1);
+	EXPECT_THAT(v1.size(), testing::Eq(1));
 
 	// Check the point is (0,0)
-	ASSERT_EQ(checkFirstOrLastPoint(0, 0, 0, v), true);	
+	EXPECT_THAT(checkFirstOrLastPoint(0, 0, 0, v1), testing::IsTrue());	
 
 	// Get the points for the line connecting (3,4) and (3,4)
-	v = bresenham_conversion(3, 4, 3, 4);
+	const auto v2 = bresenham_conversion(3, 4, 3, 4);
 
 	// Check only one point was produced
-	ASSERT_EQ(v.size(), 1);
+	EXPECT_THAT(v2.size(), testing::Eq(1));
 
 	// Check the point is (3,4)
-	ASSERT_EQ(checkFirstOrLastPoint(0, 3, 4, v), true);
+	EXPECT_THAT(checkFirstOrLastPoint(0, 3, 4, v2), testing::IsTrue());
```

Comment:
I'm going to guess that you couldn't be using this function. Also, `EXPECT_TRUE` is a gtest macro

Reply:
Done